lib/haml_lint/reporter/checkstyle_reporter.rb

```diff
@@ -0,0 +1,33 @@
+module HamlLint
+  # Outputs report as an XML checkstyle document.
+  class Reporter::CheckstyleReporter < Reporter
+    def display_report(report)
+      lints = report.lints
+      files = lints.group_by(&:filename)
+
+      xml = '<?xml version="1.0" encoding="utf-8"?>'
+
+      xml << '<checkstyle version="5.7">'
+
+      files.each do |filename, offenses|
+        xml << "<file name=\"#{filename}\">"
+        xml << render_offenses(offenses)
+        xml << '</file>'
+      end
+
+      xml << '</checkstyle>'
+      log.log xml
+    end
+
+    private
+
+    def render_offenses(offenses)
+      xml = ''
+      offenses.each do |offense|
+        xml << "<error line=\"#{offense.line}\" severity=\"#{offense.severity}\" "
+        xml << "message=\"#{offense.message}\" source=\"#{offense.linter.name}\" />"
```

Comment:
CGI::escapeHTML seems to be the recommended way to do so:
https://stackoverflow.com/questions/692921/how-to-html-encode-escape-a-string-is-there-a-built-in